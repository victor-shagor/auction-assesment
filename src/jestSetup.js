import { rest } from "msw";
import { setupServer } from "msw/node";

const mockresponse = {
  success: true,
  message: "Operation Successful",
  data: [
    {
      name: "John Doe",
      title: "Apple MacBook Air 13 M1 Chip 8GB 256GB 2020 Model - Rose Gold",
      bid: "₦795,000",
      image: "https://i.ibb.co/h7Bts6w/Imagelaptop.png",
    },
    {
      name: "Peter Doe",
      title: "Apple MacBook Air 13 M1 Chip 8GB 256GB 2020 Model - Rose Gold",
      bid: "₦795,000",
      image: "https://i.ibb.co/h7Bts6w/Imagelaptop.png",
    },
  ],
};

const handlers = [
  rest.get(
    "https://run.mocky.io/v3/7f02819f-8254-410a-b8af-ab98572bd26b",
    (req, res, ctx) => {
      return res(ctx.json(mockresponse));
    }
  ),
];

const server = setupServer(...handlers);

export { server };
