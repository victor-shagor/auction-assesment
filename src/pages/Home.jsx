import React from "react";
import Stack from "@mui/material/Stack";
import Button from "@mui/material/Button";
import Box from "@mui/material/Box";
import Divider from "@mui/material/Divider";
import { makeStyles } from "@mui/styles";
import Grid from "@mui/material/Grid";

import AuctionCard from "../components/AuctionCard";
import NavBar from "../components/Navbar";
import CountDown from "../components/CountDown";

const useStyles = makeStyles({
  wrapper: {
    marginTop: "30px",
    margin: "20px 20px",
    borderRadius: "8px",
    padding: "20px",
    border: "1px solid #EAECF0",
    fontFamily: "IBMPlexRegular"
  },
  btn: {
    textTransform: "none",
    color: "#344054",
    backgroundColor: "white",
    border: "1px solid #D0D5DD",
    borderRadius: "8px",
  },
  typo: {
    fontStyle: "normal",
    fontWeight: 60,
    fontSize: "18px",
    fontFamily: "IBMPlexSemiBold"
  },
});

const Home = () => {
  const classes = useStyles();

  return (
    <>
    <NavBar />
    <CountDown />
    <Box className={classes.wrapper}>
      <Stack
        direction="row"
        alignItems="center"
        justifyContent="space-between"
        marginBottom="10px"
      >
        <p className={classes.typo}>Featured Items</p>
        <Button
          disableElevation
          size="small"
          variant="contained"
          sx={{textTransform: "none",
          color: "#344054",
          backgroundColor: "white",
          border: "1px solid #D0D5DD",
          borderRadius: "8px"}}
        >
          View Auction
        </Button>
      </Stack>
      <Divider sx={{backgroundColor:"#EAECF0"}}/>
      <Grid container spacing={2} justifyContent="center" marginTop="5px">
        <Grid item xs={12} md={3}>
          <AuctionCard />
        </Grid>
        <Grid item xs={12} md={3}>
          <AuctionCard />
        </Grid>
        <Grid item xs={12} md={3}>
          <AuctionCard />
        </Grid>
        <Grid item xs={12} md={3}>
          <AuctionCard />
        </Grid>
      </Grid>
    </Box>
    </>
  );
};

export default Home;
