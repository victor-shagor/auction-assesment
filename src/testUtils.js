import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { productsApi } from "./services/products";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: {
        [productsApi.reducerPath]: productsApi.reducer,
      },
      preloadedState,
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(productsApi.middleware),
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
