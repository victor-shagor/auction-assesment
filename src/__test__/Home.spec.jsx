import { screen, waitFor } from "@testing-library/react";
import { rest } from "msw";
import { server } from "../jestSetup";
import Home from "../pages/Home";
import { renderWithProviders } from "../testUtils";

describe("App", () => {
  it("should render two product since mock response returns two products", async () => {
    renderWithProviders(<Home />);

    await waitFor(() => {
      expect(screen.getAllByTestId("auction-card").length).toBe(2);
      expect(screen.getByTestId("countdown")).toBeInTheDocument();
    });
  });
  it("should render a toast error if api returns an error", async () => {
    server.use(
      rest.get(
        "https://run.mocky.io/v3/7f02819f-8254-410a-b8af-ab98572bd26b",
        (_req, res, ctx) => {
          return res(ctx.status(500), ctx.json("an error has occurred"));
        }
      )
    );

    renderWithProviders(<Home />);

    await waitFor(() => {
      expect(
        screen.getByText(/Something went wrong, try again later/i)
      ).toBeInTheDocument();
    });
  });
  it("renders correctly", () => {
    const { container } = renderWithProviders(<Home />);
    expect(container).toMatchSnapshot();
  });
});
