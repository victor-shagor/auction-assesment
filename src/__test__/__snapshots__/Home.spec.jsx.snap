// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1tbrwvn"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1kdd8jn-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-i9gxme"
        >
          <p
            class="makeStyles-header-35"
          >
            Welcome
          </p>
          <span>
            Your current sales auction and activity.
          </span>
        </div>
        <span
          class="MuiBadge-root css-1c32n2y-MuiBadge-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NotificationsOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
            />
          </svg>
          <span
            class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-ujyhhw-MuiBadge-badge"
          >
            2
          </span>
        </span>
      </div>
    </header>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
  </div>
  <div
    class="makeStyles-wrapper-36 MuiBox-root css-0"
    data-testid="countdown"
  >
    <div
      class="makeStyles-innerBox-37 MuiBox-root css-0"
    >
      <img
        alt=""
        class="makeStyles-img-38"
        src="Image.png"
      />
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt=""
          class="makeStyles-logo-39"
          src="Avatarphoto.png"
        />
        <div
          class="makeStyles-countDownWrapper-40 MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="makeStyles-countdown-41"
            >
              Starts in: 3 days : 2 hours : 24 minutes
               
            </p>
            <div
              class="makeStyles-flexBox-42 MuiBox-root css-0"
            >
              <div
                class="makeStyles-liveWrapper-43 MuiBox-root css-0"
              >
                <div
                  class="makeStyles-dotBox-44 MuiBox-root css-0"
                />
                <span>
                   Not Live
                </span>
              </div>
              <span>
                Layers Auction
              </span>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation css-1y4a2p3-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ThumbUpAltOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 8h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2c0-1.1-.9-2-2-2zm0 4-3 7H9V9l4.34-4.34L12.23 10H21v2zM1 9h4v12H1z"
                />
              </svg>
            </span>
            Accept Invite
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-wrapper-33 MuiBox-root css-0"
  >
    <div
      class="MuiStack-root css-1goc8b4-MuiStack-root"
    >
      <p
        class="makeStyles-typo-34"
      >
        Featured Items
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation css-1y4a2p3-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        View Auction
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-8f17bi-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-quseuk-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="oxhzk7x-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="oxhzk7x-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#oxhzk7x-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#oxhzk7x-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="oxhzk7x-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="oxhzk7x-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="mezfe1v-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="mezfe1v-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#mezfe1v-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#mezfe1v-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="mezfe1v-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="mezfe1v-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="ir7d91f-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="ir7d91f-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#ir7d91f-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#ir7d91f-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="ir7d91f-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="ir7d91f-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="35phfik-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="35phfik-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#35phfik-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#35phfik-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="35phfik-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="35phfik-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="fyc403n-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="fyc403n-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#fyc403n-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#fyc403n-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="fyc403n-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="fyc403n-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="vvacemm-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="vvacemm-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#vvacemm-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#vvacemm-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="vvacemm-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="vvacemm-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="seqgkxl-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="seqgkxl-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#seqgkxl-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#seqgkxl-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="seqgkxl-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="seqgkxl-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        data-testid="loading-card"
      >
        <svg
          aria-labelledby="40iosvr-aria"
          height="550"
          role="img"
          viewBox="0 0 300 550"
          width="300"
        >
          <title
            id="40iosvr-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#40iosvr-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#40iosvr-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="40iosvr-diff"
            >
              <rect
                height="550"
                rx="10"
                ry="10"
                width="300"
                x="1"
                y="0"
              />
            </clippath>
            <lineargradient
              id="40iosvr-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#ecebeb"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#f3f3f3"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
